src/common/memContext.c

```diff
@@ -398,49 +399,54 @@ memContextAllocFind(void)
             // Calculate new list size
             unsigned int allocListSizeNew = contextCurrent->allocListSize * 2;
 
-            // ReAllocate memory before modifying anything else in case there is an error
-            contextCurrent->allocList = memReAllocInternal(contextCurrent->allocList, sizeof(MemContextAlloc) * allocListSizeNew);
-
-            // Initialize new allocations in list
-            for (unsigned int allocIdx = contextCurrent->allocListSize; allocIdx < allocListSizeNew; allocIdx++)
-                contextCurrent->allocList[allocIdx] = (MemContextAlloc){.active = false};
+            // Reallocate memory before modifying anything else in case there is an error
+            contextCurrent->allocList = memReAllocPtrArrayInternal(
+                contextCurrent->allocList, contextCurrent->allocListSize, allocListSizeNew);
 
             // Set new size
             contextCurrent->allocListSize = allocListSizeNew;
         }
     }
 
+    // Create new allocation
+    MemContextAlloc *result = memAllocInternal(sizeof(MemContextAlloc) + size);
+
+    *result = (MemContextAlloc)
+    {
+        .allocIdx = contextCurrent->allocFreeIdx,
+        .size = (unsigned int)(sizeof(MemContextAlloc) + size),
+    };
+
+    // Set pointer in allocation list
+    contextCurrent->allocList[contextCurrent->allocFreeIdx] = result;
+
+    // Update free index to next location. This location may not actually be free but it is where the search should start next time.
     contextCurrent->allocFreeIdx++;
 
-    // Return buffer
-    FUNCTION_TEST_RETURN(&contextCurrent->allocList[contextCurrent->allocFreeIdx - 1]);
+    FUNCTION_TEST_RETURN(result);
 }
 
 /***********************************************************************************************************************************
-Find a memory allocation
+Resize memory that has already been allocated
 ***********************************************************************************************************************************/
-static unsigned int
-memFind(const void *buffer)
+static MemContextAlloc *
+memContextAllocResize(MemContextAlloc *alloc, size_t size)
 {
     FUNCTION_TEST_BEGIN();
-        FUNCTION_TEST_PARAM_P(VOID, buffer);
+        FUNCTION_TEST_PARAM(SIZE, size);
     FUNCTION_TEST_END();
 
-    ASSERT(buffer != NULL);
+    // Make sure the allocation is in the memory context
+    ASSERT(memContextStack[memContextCurrentStackIdx].memContext->allocList[alloc->allocIdx] == alloc);
```

Comment:
I don't think we want a CHECK here because that would be run in production and this is a very hot path.

However, I did collect a bunch of ASSERTs together in a single macro, ASSERT_ALLOC_VALID(), that can be run during development to catch issues.  Mem contexts have the same weakness RE validation but we haven't seen issues with them outside development so I think the added performance is worth the trade-off here.